### PR TITLE
DBZ-2698 Added support for null default value for types int, bigint, smallint, tinyint, float, real

### PR DESCRIPTION
**Pull request by zxxz**
https://issues.redhat.com/browse/DBZ-2698

DBZ-2698 Added support for null default value for types int, bigint, smallint, tinyint, float, real

### CODE REVIEW

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerDefaultValueConverter.java [hunk 1]
```diff
@@ -118,19 +118,19 @@ private Object convertDefaultValue(Object defaultValue, Column column) {
         final Map<String, DefaultValueMapper> result = new HashMap<>();
 
         // Exact numbers
-        result.put("bigint", v -> Long.parseLong(v.substring(2, v.length() - 3))); // Sample value: ((3147483648.))
-        result.put("int", v -> Integer.parseInt(v.substring(2, v.length() - 2))); // Sample value: ((2147483647))
-        result.put("smallint", v -> Short.parseShort(v.substring(2, v.length() - 2))); // Sample value: ((32767))
-        result.put("tinyint", v -> Short.parseShort(v.substring(2, v.length() - 2))); // Sample value: ((255))
+        result.put("bigint", v -> nullableDefaultValueMapper(v, value -> Long.parseLong(value.replaceAll("(\\.)$", "")))); // Sample value: ((3147483648.))
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerDefaultValueConverter.java:121]: Using an inline regexp isn't ideal for this one. At the very least, this should be pre-compiled into a `Pattern`, but actually, there's a better way:

    Long.parseLong(value.charAt(lastIndex) == '.' ? value.substring(0, lastIndex) : value)

It's an order of magnitude faster as per a quick JMH benchmark:

```
Benchmark                                  Mode  Cnt          Score          Error  Units
TrimBenchmark.testCompiledRegexp          thrpt    3   46507099,171 ±  3115790,518  ops/s
TrimBenchmark.testRegexp                  thrpt    3   24495251,001 ± 18640458,291  ops/s
TrimBenchmark.testSubstringEndsWith       thrpt    3  286420819,975 ± 38003048,105  ops/s
TrimBenchmark.testSubstringCharAt         thrpt    3  301351671,658 ± 86578981,123  ops/s
```

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerDefaultValueConverter.java [hunk 2]
```diff
@@ -178,4 +178,14 @@ private Object convertDefaultValue(Object defaultValue, Column column) {
         // Other data types, such as cursor, xml or uniqueidentifier, have been omitted.
         return result;
     }
+
+    public static Object nullableDefaultValueMapper(String v, DefaultValueMapper mapper) throws Exception {
+        final String value = v.replaceAll("^[\\(]+|[\\)]+$", ""); // trim leading and trailing parenthesis
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerDefaultValueConverter.java:183]: See above; let's rather use subString().

File: debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectionIT.java [hunk 3]
```diff
@@ -250,6 +250,131 @@ public void shouldProperlyGetDefaultColumnValues() throws Exception {
...
+                    + ");";
+
+            connection.execute(sql);
+
+            // then enable CDC and wrapper functions
+            TestHelper.enableTableCdc(connection, "table_with_defaults");
+            // insert some data
+
+            // and issue a test call to a CDC wrapper function
+            Thread.sleep(5_000); // Need to wait to make sure the min_lsn is available
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectionIT.java:309]: Let's rather use Awaitility with a defined condition (see other tests for examples).

**Review by gunnarmorling — COMMENTED** [review]
Thanks! A few comments inline.

**Comment by gunnarmorling** [comment]
Hey @zxxz, just checking in on whether you'd still like to wrap this one up as per above comments? Thanks!

**Comment by gunnarmorling** [comment]
@zxxz, could you give an indication whether you'd like to finish this, or whether someone else should take over? Thx!

**Comment by zxxz** [comment]
> @zxxz, could you give an indication whether you'd like to finish this, or whether someone else should take over? Thx!

Hello @gunnarmorling, yes I would like to finish this! Give me a few days though as I have to finish some other stuff first...

**Comment by gunnarmorling** [comment]
Hey @zxxz, any news here?

**Comment by gunnarmorling** [comment]
Applied; thanks a lot for driving this home, @zxxz!

**Pull request merged**